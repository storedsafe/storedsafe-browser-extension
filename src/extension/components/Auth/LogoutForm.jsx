import React from 'react';
import PropTypes from 'prop-types';
import Input from '../Input';

function LogoutForm({ onLogout }) {
  const onSubmit = (event) => {
    event.preventDefault();
    onLogout();
  };

  return (
    <form className="logout" onSubmit={onSubmit}>
      <Input type="submit" name="submit" title="Logout from StoredSafe" />
    </form>
  );
}

LogoutForm.propTypes = {
  onLogout: PropTypes.func.isRequired,
};

export default LogoutForm;
