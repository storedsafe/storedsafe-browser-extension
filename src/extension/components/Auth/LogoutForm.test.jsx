import React from 'react';
import { shallow, mount } from 'enzyme';
import LogoutForm from './LogoutForm';

const onLogout = jest.fn();

describe('rendering', () => {
  test('<LogoutForm />', () => {
    const wrapper = shallow(<LogoutForm onLogout={onLogout} />);
    expect(wrapper.find('Input[type="submit"]').length).toBe(1);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('interaction', () => {
  test('<LogoutForm />', () => {
    const wrapper = mount(<LogoutForm onLogout={onLogout} />);
    wrapper.simulate('submit');
    expect(onLogout).toHaveBeenCalled();
  });
});
