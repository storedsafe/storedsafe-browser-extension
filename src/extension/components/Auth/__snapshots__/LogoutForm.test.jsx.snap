// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering <LogoutForm /> 1`] = `ShallowWrapper {}`;
